src/messages/tests/integration.rs

```diff
@@ -0,0 +1,749 @@
+        signing_commitments: big_signing_commitments,
+        message: "hola".as_bytes().to_vec(),
+    });
+    let validate_payload = Validate::validate(&payload);
+    assert_eq!(validate_payload, Err(MsgErr::TooManyCommitments));
+
+    // change to 2 commitments
+    let commitments = vec![commitment1[0], commitment2[0]];
+    let participants = vec![setup.signer1, setup.signer2];
+    let signing_commitments = create_signing_commitments(commitments, participants);
+
+    let big_message = [0u8; constants::ZCASH_MAX_PROTOCOL_MESSAGE_LEN + 1].to_vec();
+    let payload = Payload::SigningPackage(SigningPackage {
+        signing_commitments,
+        message: big_message,
+    });
+    let validate_payload = Validate::validate(&payload);
+    assert_eq!(validate_payload, Err(MsgErr::MsgTooBig));
+
+    let message = Message {
+        header,
+        payload: payload.clone(),
+    };
+    let validate_message = Validate::validate(&message);
+    assert_eq!(validate_message, Err(MsgErr::SenderMustBeAggregator));
+
+    // change header
+    let header = create_valid_header(setup.aggregator, setup.dealer);
+
+    let message = Message {
+        header: header,
+        payload: payload.clone(),
+    };
+
+    let validate_message = Validate::validate(&message);
+    assert_eq!(validate_message, Err(MsgErr::ReceiverMustBeSigner));
+
+    let header = create_valid_header(setup.aggregator, setup.signer1);
+
+    let validate_message = Validate::validate(&Message { header, payload }).err();
+    assert_eq!(validate_message, None);
+}
+
+#[test]
+fn serialize_signingpackage() {
+    let mut setup = basic_setup();
+
+    let (_nonce, commitment1) = frost::preprocess(1, u64::from(setup.signer1), &mut setup.rng);
+    let (_nonce, commitment2) = frost::preprocess(1, u64::from(setup.signer2), &mut setup.rng);
+
+    let header = create_valid_header(setup.aggregator, setup.signer1);
+
+    let commitments = vec![commitment1[0], commitment2[0]];
+    let participants = vec![setup.signer1, setup.signer2];
+    let signing_commitments = create_signing_commitments(commitments, participants);
+
+    let payload = Payload::SigningPackage(SigningPackage {
+        signing_commitments: signing_commitments.clone(),
+        message: "hola".as_bytes().to_vec(),
+    });
+
+    let message = Message {
+        header: header,
+        payload: payload.clone(),
+    };
+
+    // check general structure serialization/deserialization
+    serialize_message(message, setup.aggregator, setup.signer1);
+
+    // check payload serialization/deserialization
+    let mut payload_serialized_bytes = bincode::serialize(&payload).unwrap();
+    // remove the padding. TODO: why is this happening ?
+    payload_serialized_bytes =
+        (&payload_serialized_bytes[4..payload_serialized_bytes.len()]).to_vec();
+
+    let deserialized_map_len: u32 = bincode::deserialize(&payload_serialized_bytes[0..8]).unwrap();
+    assert_eq!(deserialized_map_len, 2);
```

Comment:
Ah, this is very similar to what I suggested before, but here it is using the 8-byte length for the map. Maybe it's a good idea to change the RFC so that it's 8 bytes (`u64`)?

Reply:
Yea, this seems too complicated to implement manually now. I changed the `u32` from that line and added length check in another test that uses the map too https://github.com/ZcashFoundation/redjubjub/pull/114/commits/503af0e6667d21b859636e954599ba4b8335e327

I am unsure if we should change the spec or if we should try to make it `u8` in the future and save some bytes. We know the len will never be bigger than an `u8` so the rest is unnecessary.

Reply:
https://github.com/ZcashFoundation/redjubjub/issues/118